### PR TITLE
Add wrapper for Nevergrad PSO optimizer

### LINKED ISSUE #560: Add wrappers for Nevergrad
Wrap the gradient free optimizers from [nevergrad](https://github.com/facebookresearch/nevergrad) in optimagic.

Nevergrad implements the following algorithms:

- `NgIohTuned` is "meta"-optimizer which adapts to the provided settings (budget, number of workers, parametrization) and should therefore be a good default.
- `TwoPointsDE` is excellent in many cases, including very high `num_workers`.
- `PortfolioDiscreteOnePlusOne` is excellent in discrete settings of mixed settings when high precision on parameters is not relevant; it's possibly a good choice for hyperparameter choice.
- `OnePlusOne` is a simple robust method for continuous parameters with `num_workers` < 8.
- `CMA` is excellent for control (e.g. neurocontrol) when the environment is not very noisy (num_workers ~50 ok) and when the budget is large (e.g. 1000 x the dimension).
- `TBPSA` is excellent for problems corrupted by noise, in particular overparameterized (neural) ones; very high `num_workers` ok).
- `PSO` is excellent in terms of robustness, high `num_workers` ok.
- `ScrHammersleySearchPlusMiddlePoint` is excellent for super parallel cases (fully one-shot, i.e. `num_workers` = budget included) or for very multimodal cases (such as some of our MLDA problems); don't use softmax with this optimizer.
- `RandomSearch` is the classical random search baseline; don't use softmax with this optimizer.

In the long run we want to wrap all of them, **but if you are tackling this as your first issue you should focus on one. In that case please coment below which optimizer you are going to work on so we don't duplicate efforts.**

### PR DESCRIPTION
**Pull request by r3kste**
Implements the Particle Swarm Optimization algorithm from `nevergrad`.
Partially fixes #560

### CODE REVIEW

**Comment by r3kste** [comment]
Hello @janosg Is there anything left to implement in this PR? I would appreciate it if the CI gets approval to begin.

**Comment by gulshan-123** [comment]
@r3kste , I am also working on similar PR: about the addition of OnePlusOne algorithm from nevergrad. I am curious if we can also add non-linear constraint in wrapper as suggested here: https://facebookresearch.github.io/nevergrad/optimization.html#optimization-with-constraints, can you please look into this?

File: src/optimagic/optimizers/nevergrad_optimizers.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            qo=self.quasi_opp_init,
+            sqo=self.speed_quasi_opp_init,
+            so=self.special_speed_quasi_opp_init,
+        )(
+            parametrization=instrum,
+            budget=self.stopping_maxfun,
+            num_workers=self.n_cores,
+        )
+
+        while optimizer.num_ask < optimizer.budget:
```
Comment (janosg) [inline src/optimagic/optimizers/nevergrad_optimizers.py:82]: Minor thing, but I think it would be more readable to see `while optimizer.num_ask <= self.stopping_maxfun`

File: src/optimagic/optimizers/nevergrad_optimizers.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            for x, loss in zip(x_list, losses, strict=True):
+                optimizer.tell(x, loss)
+
+        recommendation = optimizer.provide_recommendation()
+
+        result = InternalOptimizeResult(
+            x=recommendation.value[0][0],
+            fun=recommendation.loss,
+            success=True,
+            n_fun_evals=optimizer.num_ask,
```
Comment (janosg) [inline src/optimagic/optimizers/nevergrad_optimizers.py:101]: Can you try to define a few more fields of the `InternalOptimizeResult`, even if they are trivial (e.g. `n_jac_evals` could be 0 instead of None).

File: src/optimagic/optimizers/nevergrad_optimizers.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            parametrization=instrum,
+            budget=self.stopping_maxfun,
+            num_workers=self.n_cores,
+        )
+
+        while optimizer.num_ask < optimizer.budget:
+            x_list = [
+                optimizer.ask()
+                for _ in range(
+                    min(optimizer.num_workers, optimizer.budget - optimizer.num_ask)
```
Comment (janosg) [inline src/optimagic/optimizers/nevergrad_optimizers.py:86]: Similarly, I would prefer `self.n_cores` over `optimizer.num_workers` here. For readers who are familiar with optimagic but not nevergrad (which is going to be the majority) it is easier to understand the optimagic terminology than the nevergrad terminology.

**Review by janosg — COMMENTED** [review]
Hi @r3kste, thanks for the PR. The changes look really good. I just made a few minor stylistic comments. 

Currently the tests are not passing but I think this is not related to a problem in your code. It rather seems like nevergrad has a few things that are not yet compatible with numpy 2.0. 

Can you pin numpy to <2.0 in the environment to confirm this? If it works we can see how to deal with the situation.

**Review by janosg — APPROVED** [review]
Thanks a lot for the great PR!
